martin/src/file_config.rs

```diff
@@ -168,8 +168,18 @@ impl<T: ConfigExtras> FileConfigEnum<T> {
 
     pub fn finalize(&self, prefix: &str) -> UnrecognizedValues {
         let mut res = UnrecognizedValues::new();
+
         if let Self::Config(cfg) = self {
             copy_unrecognized_config(&mut res, prefix, cfg.get_unrecognized());
+
+            if !res.is_empty() {
+                warn!(
+                    "Unrecognized configuration values found in prefix '{}': {:?}. \
+                    These will be ignored. Please check your configuration file for typos or deprecated keys.",
+                    prefix,
+                    cfg.get_unrecognized()
+                );
+            }
```

Comment:
Much better.

I think below code could be a bit more user friendly.
This way, we are telling people less abstract which value is unexpected.

This also fixes the duplicate reporting bug caused by checking `res` instead of `unrecognized`.

```suggestion
            let unrecognized = cfg.get_unrecognized();
            copy_unrecognized_config(&mut res, prefix, unrecognized);

            for (key, value) in unrecognized.into_iter() {
                warn!("Ignoring unrecognized configuration entry '{prefix}.{key}: {value}'. Please check your configuration file for typos.");
            }
```

Reply:
Thanks @CommanderStorm .